onnxruntime/core/framework/session_options.cc

```diff
@@ -57,4 +57,13 @@ Status SessionOptions::AddExternalInitializers(gsl::span<const std::string> name
   return Status::OK();
 }
 #endif  // !defined(ORT_MINIMAL_BUILD) && !defined(DISABLE_EXTERNAL_INITIALIZERS)
+
+void SessionOptions::AddCustomOpLibraryHandle(void* library_handle, const char* library_name) {
+  if (!custom_op_libs_) {
+    custom_op_libs_ = std::make_shared<std::vector<LibraryHandle>>();
```

Comment:
We can also create this shared pointer in the SessionOptions constructor, which does not currently exist. If we add the constructor, we should remember to add default implementations for copy/move/destructor/etc.